patches/api/0429-Expose-hand-during-BlockCanBuildEvent.patch

```diff
@@ -0,0 +1,48 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: The456gamer <the456gamer@the456gamer.dev>
+Date: Mon, 21 Aug 2023 14:13:43 +0100
+Subject: [PATCH] Expose hand during BlockCanBuildEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/block/BlockCanBuildEvent.java b/src/main/java/org/bukkit/event/block/BlockCanBuildEvent.java
+index a1350c0f74d445dca09eea6e10abac050bb06990..dab832a3c2ab4319c31cd5c35b1bbb1aab4e671a 100644
+--- a/src/main/java/org/bukkit/event/block/BlockCanBuildEvent.java
++++ b/src/main/java/org/bukkit/event/block/BlockCanBuildEvent.java
+@@ -25,10 +25,11 @@ public class BlockCanBuildEvent extends BlockEvent {
+ 
+     protected BlockData blockData;
+     private final Player player;
++    private final org.bukkit.inventory.EquipmentSlot hand; // Paper - expose hand
+ 
+     @Deprecated
+     public BlockCanBuildEvent(@NotNull final Block block, @NotNull final BlockData type, final boolean canBuild) {
+-        this(block, null, type, canBuild);
++        this(block, null, type, canBuild, org.bukkit.inventory.EquipmentSlot.HAND); // Paper - expose hand
+     }
+ 
+     /**
+@@ -37,12 +38,23 @@ public class BlockCanBuildEvent extends BlockEvent {
+      * @param type the id of the block to place
+      * @param canBuild whether we can build
+      */
+-    public BlockCanBuildEvent(@NotNull final Block block, @Nullable final Player player, @NotNull final BlockData type, final boolean canBuild) {
++    public BlockCanBuildEvent(@NotNull final Block block, @Nullable final Player player, @NotNull final BlockData type, final boolean canBuild, @NotNull final org.bukkit.inventory.EquipmentSlot hand) { // Paper - expose hand
```

Comment:
Add `@ApiStatus.Internal` to the constructor. It's possible we want this to be a new constructor. Usually for most events I don't care, but its possible this event is created/fired by plugins more than others.